### ISSUE TITLE
How to specify options in the request?

### ISSUE THREAD

**Issue by veslav3** [cyclosproject/ng-swagger-gen#77]
Normally I would create a request like this:

```JS
const req = new HttpRequest('POST', `URL_HERE`, formData, {reportProgress: true});
```

Now I'd do this with the generated code:
```JS
this.uploadService.uploadPreviewUsingPOSTResponse(previewFile)
```

But I want to see the upload progress so I can display it in the bar like this:
```JS
.subscribe(event => {
      if (event.type === HttpEventType.UploadProgress) {
        this.previewUploadPercentage = `${Math.round(100 * event.loaded / event.total)}%`;
        console.log(`File is ${this.previewUploadPercentage}% uploaded.`);
      } else if (event instanceof HttpResponse) {
        this.previewId = event.body['id'];
      }
```

How could I use the result of the reportProgress option?

**Comment by luisfpg** [comment]
This is not supported out-of-the-box, but would be nice to have.
Luckily there is a workaround: create an [http interceptor](https://github.com/cyclosproject/ng-swagger-gen#passing-request-headers--customizing-the-request) that adds the request options.

**Comment by luisfpg** [comment]
I was thinking about this, but I think this doensn't fit the general philosophy of the generated services.
If you look at each generated service, we make sure that the event is filtered to be an instanceo of 
`HttpResponse` (in the `<method>Response` methods) and then extract the response body in `<method>`.
If we add `reportProgress`, then the returned stream should really be of HttpEvent, which, in turn, would either require a third form for each generated method (not desirable), and we would need to receive a partial list of options.
So, to summarize, for the current phiosophy, I'd say this is not desirable. For the specific requests you need to report progress, I'd say you should manually perform the requests.
If you want to discuss it further, the issue can be reopened.

**Comment by veslav3** [comment]
@luisfpg we do the request manually now, but I'm still not giving up on finding another way around the issue. :)

**Comment by luisfpg** [comment]
Of course, @veslav3, we're always open to clever ideas.
